syntax = "proto3";

service FileServer {
    rpc DownloadFile(DownloadFileRequest) returns (DownloadFileResponse) {}
	rpc SaveFile(SaveFileRequest) returns (SaveFileResponse) {}
    rpc ListFiles(ListFilesRequest) returns (ListFilesResponse) {}
}

message DownloadFileRequest {
    string file_name = 1;
}

message DownloadFileResponse {
    bool success = 1;
    string message = 2;
    bytes file_content = 3;
}

message SaveFileRequest {
    string file_name = 1;
    bytes file_content = 2;
}

message SaveFileResponse {
    bool success = 1;
    string message = 2;
}

message ListFilesRequest {
}

message ListFilesResponse {
    bool success = 1;
    repeated string fileName = 2;
}